### PR TITLE
News Api installation

### PR DESCRIPTION
**Pull request by Noelryn**
## PT:
https://www.pivotaltracker.com/story/show/161220378

## Included in PR 
- implementing News API
- Installed news api gem
- Created a service folder and service file
- Created controller for News API
- Feature tests for API
## What we learned
- How to write implementation code for an API
- Change and hide our credentials 
- How to add what we need to a view from an API
## Screenshot
screen shot 2018-10-21 at 11 36 37

### CODE REVIEW

File: features/support/env.rb [hunk 1]
```diff
@@ -28,3 +29,14 @@
 Capybara.javascript_driver = :chrome
 
 World(FactoryBot::Syntax::Methods)
+Cucumber::Rails::Databse.javascript_strategy = :truncation
+WebMock.disable_net_connect!(allow_localhost: true)
+Before do
+  stub_request(:get, https://newsapi.org/v2/everything?q=bitcoin&apiKey=62aeaa7f151d4e21a0d8bcc6032044ea ).
```
Comment (tochman) [inline features/support/env.rb]: This needs to be a string @Noelryn

File: features/support/env.rb [hunk 2]
```diff
@@ -28,3 +29,14 @@
...
+  with(
+    headers: {
+      'Accept'=>'*/*',
+      'Accept-Edcoding'=>'gzip, deflate',
+    }
+  )
```
Comment (tochman) [inline features/support/env.rb]: You need a `and_return` Please just copy what Webmock gives you in the terminal

File: features/support/env.rb (hunk 1 above)
Comment (Noelryn) [reply features/support/env.rb]: Got it, I am currently in the process for fixing this

File: features/support/env.rb (hunk 2 above)
Comment (Noelryn) [reply features/support/env.rb]: I am on that too, just fixing it now

File: .gitignore [hunk 3]
```diff
@@ -16,3 +16,9 @@
 .byebug_history
 
 /config/master.key
+
+# Ignore master key for decrypting credentials and more.
+/config/master.key
```
Comment (tochman) [inline .gitignore:19]: Please look at line 18

Comment (Noelryn) [reply .gitignore:19]: Just saw this and fixed the issue, should have seen that apologies.

File: app/views/partials/_breaking.html.erb [hunk 4]
```diff
@@ -5,7 +5,7 @@
     <div class="mb-8">
         <h2 class="text-black font-bold text-xl mb-2 my-4 article-title">Article Title 1</h2>            
         <p class="article-paragraph article-content">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque laoreet posuere mi, in eleifend orci elementum eu. Vestibulum id pellentesque lorem, vel ultrices mauris. Vivamus pretium placerat orci, eu varius velit vestibulum et. Sed in libero eu orci imperdiet semper. Curabitur in metus nunc. Mauris accumsan maximus turpis, non ullamcorper risus vulputate sit amet. Etiam dictum arcu id tempus tempor. Nam suscipit cursus ante nec vehicula. Nulla neque orci, venenatis nec auctor sed, volutpat eu sapien. Vestibulum in convallis dui, vel fermentum risus. Aliquam et dui vitae ligula viverra auctor vitae id sem. In arcu libero, cursus non libero eu, aliquet fringilla velit.iption</p>
-        <p class="mt-2"> By: Donald Duck</p>
+        <p class="mt-2 top-article-agency"> By: Donald Duck</p>
```
Comment (tochman) [inline app/views/partials/_breaking.html.erb]: This one is not even used @Noelryn

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
+  resources :api_feed, only: [:index]
```
Comment (Rinbo) [inline config/routes.rb]: Do we really need this? It seems to me we are only using the main_controller together with the api-service.

File: app/controllers/api_feed_controller.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class ApiFeedController < ApplicationController
+
+  def index
+    @apifeed = FeedService.get_feed('politics', 'bbc-news', 'bbc.co.uk', 10)
+    @top_article = @apifeed.first
+  end
+end
```
Comment (Rinbo) [inline app/controllers/api_feed_controller.rb]: Do we need this controller at all? Where are we using it?

**Pull request merged**